stathat.go

```diff
@@ -104,6 +119,35 @@ func NewReporter(bufferSize, poolSize int, transport http.RoundTripper) *Reporte
 	return r
 }
 
+type statCache struct {
+	counterStats map[string]int
+	valueStats   map[string][]float64
+}
+
+func (sc *statCache) AverageValue(statName string) float64 {
+	total := 0.0
+	values := sc.valueStats[statName]
```

Comment:
Yeah I don't think that case can happen, but doesn't hurt to be safe with that, I'll add it.